packages/backend/src/modules/tvl/TvlModule.ts

```diff
@@ -40,113 +29,81 @@ export function createTvlModule(
+    priceRepository: new PriceRepository(database, logger),
+    balanceRepository: new BalanceRepository(database, logger),
+    reportRepository: new ReportRepository(database, logger),
+    aggregatedReportRepository: new AggregatedReportRepository(
+      database,
+      logger,
+    ),
+    reportStatusRepository: new ReportStatusRepository(database, logger),
+    balanceStatusRepository: new BalanceStatusRepository(database, logger),
+  }
   // #endregion
+
   // #region peripherals
 
-  const ethereumProvider = new providers.AlchemyProvider(
-    'mainnet',
-    config.tvl.alchemyApiKey,
-  )
-  const ethereumClient = new EthereumClient(ethereumProvider, logger)
-  const multicall = new MulticallClient(ethereumClient)
   const coingeckoClient = new CoingeckoClient(http, config.tvl.coingeckoApiKey)
   const coingeckoQueryService = new CoingeckoQueryService(coingeckoClient)
-  const etherscanClient = new EtherscanClient(
-    http,
-    config.tvl.etherscanApiKey,
-    logger,
-  )
-  const ethereumBalanceProvider = new EthereumBalanceProvider(multicall)
 
   // #endregion
+
   // #region updaters
 
-  const blockNumberUpdater = new BlockNumberUpdater(
-    etherscanClient,
-    blockNumberRepository,
-    clock,
-    logger,
-    ChainId.ETHEREUM,
-  )
   const priceUpdater = new PriceUpdater(
     coingeckoQueryService,
-    priceRepository,
+    db.priceRepository,
     clock,
     config.tokens,
     logger,
   )
-  const balanceUpdater = new BalanceUpdater(
-    ethereumBalanceProvider,
-    blockNumberUpdater,
-    balanceRepository,
-    balanceStatusRepository,
-    clock,
-    config.projects,
-    logger,
-    ChainId.ETHEREUM,
-  )
-  const reportUpdater = new ReportUpdater(
-    priceUpdater,
-    balanceUpdater,
-    reportRepository,
-    aggregatedReportRepository,
-    reportStatusRepository,
-    clock,
-    config.projects,
-    logger,
-  )
 
   // #endregion
+
   // #region api
 
-  const blocksController = new BlocksController(blockNumberRepository)
+  const blocksController = new BlocksController(db.blockNumberRepository)
   const tvlController = new TvlController(
-    reportStatusRepository,
-    aggregatedReportRepository,
-    reportRepository,
+    db.reportStatusRepository,
+    db.aggregatedReportRepository,
+    db.reportRepository,
     config.projects,
     config.tokens,
     logger,
   )
 
-  const dydxController = new DydxController(aggregatedReportRepository)
+  const dydxController = new DydxController(db.aggregatedReportRepository)
 
   const blocksRouter = createBlocksRouter(blocksController)
   const tvlRouter = createTvlRouter(tvlController)
   const dydxRouter = createDydxRouter(dydxController)
 
   // #endregion
 
-  const start = async () => {
-    if (!config.syncEnabled) {
-      logger.info('TVL sync disabled')
-      return
-    }
+  const modules: (ApplicationModule | undefined)[] = [
+    createEthereumTvlModule(db, priceUpdater, config, logger, http, clock),
```

Comment:
WYT about renaming this to `Submodule`? It's one step deeper than module, so would be less confusing.
```suggestion
    createEthereumTvlSubmodule(db, priceUpdater, config, logger, http, clock),
```

Reply:
I like that idea! I've renamed these functions